### ISSUE TITLE
[Bug]: Token footprint is kept when changing to native size

### ISSUE THREAD

**Issue by kwvanderlinde** [RPTools/maptool#4856]
### Describe the Bug

If a _Medium_ token (1x1 cell) is changed to _Native_ size, the footprint stays as one cell. If a _Gargantuan_ token (4x4 cells) is changed to _Native_ size, its footprint stays as 4x4 cells despite the user seeing no reference to a _Gargantuan_ size.

### To Reproduce

1. Drag two tokens onto a map with square grid 
2. Change one token to size _Medium_ and the other to size _Gargantuan_
3. Change both tokens to size _Native_.
4. Drag each token. Notice the originally-medium token still occupies one cell, while the originally-Gargantuan token occupies 16 cells

### Expected Behaviour

The token footprint should be based on its actual size. If the token visibly covers 9 cells, that should be its footprint. Alternatively, natively-sized tokens could simply occupy one cell at its center.

### Screenshots

This shows the two apparently-identical tokens with their differing footprints:
![image](https://github.com/RPTools/maptool/assets/7492219/68ff0e8c-6ec1-4141-b77b-e6ce1413c744)

### MapTool Info

1.14.3

### Desktop

Linux Mint 21.3

**Comment by bubblobill** [comment]
- Native size footprints should probably be the size of the drawn image instead of one cell.
- I can actually see benefits in this if it was something you could do deliberately. e.g. A token has an effect that covers a certain footprint. You could then use macros to apply the effect on other tokens within the footprint.
- We really need a footprint editor dialogue

**Comment by bubblobill** [comment]
No. What you see is not necessarily related to the occupied space. 
WRT retention of footprint. That should just need `isSnapToScale` being set to false when it is changed to free-size
Between footprints changing, where do we need to update the occupied cells cache?
